src/target/generated.rs

```diff
@@ -14,6 +14,7 @@ pub(crate) const LIST: &[(&str, TargetInfo<'static>)] = &[
             env: "",
             abi: "",
             unversioned_llvm_target: "arm64-apple-macosx",
+            features: "aes,crc,dit,dotprod,dpb,dpb2,fcma,fhm,flagm,flagm2,fp16,frintts,jsconv,lor,lse,lse2,neon,paca,pacg,pan,pmuv3,ras,rcpc,rcpc2,rdm,sb,sha2,sha3,ssbs,v8.1a,v8.2a,v8.3a,v8.4a,vh",
```

Comment:
The generated file is getting larger and I'm slightly worried about the compile-time/binary size.

It probably would be fine for now and we should not hesitate adding more, if it eases maintenance burden and reduce hardcoded values.

Reply:
I suspect that the features here are going to be fairly important, but yeah, I do understand your worry.

I guess if we wanted to decrease binary size, we could parse them into a `&[Feature]` beforehand?

Reply:
On the other hand, we could replace all this features with a boolean since all we need to know is if `cry-static` is included in the feature or not.

If we need more features later, then we can introduce them later, and probably use an enum which is more efficient than `&str`

Reply:
If we want to reduce the binary size/compilation time, then I could think of two ideas:

First, split the target and `TargetInfo` into two arrays, this would help simplify the array, and might make binary search faster as more target can be in the cache.

For `TargetInfo`, we can store the string inline.

We can internalise the string into a one big `&str` and refer to it via index, we can simply use a `u16` for base pointer.

The simplest impl is to use `HashSet` in gen-target-info to dedup them, and then concat them into one giant string, each terminated by `\0` and assign them each a `u16` based on the length within the giant string.

That will compress 16B `&'static str` on 64-bit platform to 2B, and should improve compile-time as well.

Though it looks like something out of scope for this ticket.

Reply:
So for this PR, cc @madsmtm I just want you to replace `features: &str` with a `crt_static: boolean`, that would be good enough.

The rest of the idea should be discussed and completed in a different PR.